### ISSUE TITLE
useSWRInfinite - calling setSize refetches first page on every call

### ISSUE THREAD

**Issue by CGreenburg** [vercel/swr#1401]
# Bug report

## Description / Observed Behavior

Running the infinite-scroll example on the latest release (1.0.0), I observe that the first page gets re-fetched on every call to `setSize(size + 1)`

## Expected Behavior

`setSize(size +1)` should only fetch the next page.

## Repro Steps / Code Example

1. Clone master
2. Add `console.log(...args)` to `examples/infinite-scroll/libs/fetch.js`
3. `npm install && npm run dev`
4. Inspect console logs as you scroll.

## Additional Context

Changing the version of swr in the example back to 0.5.6 and updating the import of useSWRInfinite , this doesn't happen.

**Comment by shuding** [comment]
This is the expected behavior, we should have made it clear on the docs 🙏 . I’ll write to explain later.

**Comment by CGreenburg** [comment]
So when fetching the next page, the expected behavior is to make two requests for the first and next page? Is there a way to disable this?

My concern is that if I implement 1.0.0, my infinite scrolling component is now sending double the requests for every user on the website.

**Comment by shuding** [comment]
@CGreenburg Yes. The reason behind this is SWR is unsure about if the next cursor is stable.

Imagine a timeline-like (Twitter/Facebook/GitHub) application, where new items will be appended to the _first_ page. Now if there’re 3 pages loaded already, when hitting “Load Next”, SWR doesn’t know if there’s any new item in the first page

**Comment by CGreenburg** [comment]
That makes sense why the first page would be refetched. 

Using Reddit infinite-scroll as an example: if there are three pages and the fourth is asked for, the first page needs revalidation. If the first and second pages change, the user might scroll back up to revisit a post and see something they hadn’t scrolled past earlier. It could also cause a content shift if something in their viewport changes because of revalidation.

But you you also brought up a good point about duplicates if revalidation doesn’t occur.

Thanks for walking me though this. I think it comes down to how the backend systems are serving pages to begin with. Because of that I don’t think we need to keep this open because SWR’s behavior makes sense.

**Comment by msnegurski** [comment]
Would still be great if there was an ability to disable this behaviour. Cursor based pagination kind of already takes care about it, and in some usecases it doesn't make sense. If I am on 20th page, useSWRinfinite makes 20 requests before loading next page, which could take a lot of time.

Reading the documentation, I was thinking that `revalidateAll` flag is what disables this behaviour, but it either doesn't work or is used for something else.

**Comment by sanders-vag** [comment]
I am also interested in disabling this behavior, could this be considered? Thanks!!!

**Comment by shuding** [comment]
I reopened and changed it to a feature request. 👍  If the API is cursor based it makes sense to have this option.

**Comment by tomaszczura** [comment]
I am interested on that too... We have items which won't be changed as often as reddit, but loading first page over and over again will lead to a bad UX - when user sees a loader, we have to load first page (which often is not visible), and then the one user wants to see - so user has to wait twice longer than it's necessary.

**Comment by shuding** [comment]
@easyrun32 It's included in the latest beta, not 1.0.0. Please run `npm i swr@beta` to upgrade.
